qemu/tests/pci_bridge_hotplug_unplug.py

```diff
@@ -0,0 +1,160 @@
+    """
+    pci_bridge_num = int(params.get("pci_bridge_num", 1))
+    if pci_bridge_num > 1:
+        raise NotImplementedError()
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    login_timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=login_timeout)
+    cmd_timeout = int(params.get("cmd_timeout", 3600))
+
+    image_list = params.objects("images")
+    del image_list[0]
+    image_num = len(image_list)
+    if (image_num < 1):
+        test.cancel("No pre-prepared images for hotplugging!")
+
+    device_list = []
+    fmt_list = params.objects("disk_driver")
+    if pci_bridge_num < 1:
+        test.cancel("There should be at least one pci-bridge!")
+    elif pci_bridge_num == 1:
+        pci_bridge_id = params["pci_controllers"]
+        for i in range(image_num):
+            if i >= len(fmt_list):
+                test.error("No sufficient disk driver type!")
+            error_context.context("Hotplug a %s disk!" % fmt_list[i],
+                                  logging.info)
+            device_list += disk_hotplug(test, params, vm, session,
+                                        image_list[i], fmt_list[i],
+                                        pci_bridge_id, cmd_timeout)
+    else:
+        raise NotImplementedError()
+
+    error_context.context("Check hotplugged disks in monitor!", logging.info)
+    monitor_info_block = vm.monitor.info_block(False)
+    blocks = ','.join(monitor_info_block.keys())
+    for image in image_list:
+        if image not in blocks:
+            test.fail("drive_%s is missed: %s!" % (image, blocks))
+
+    error_context.context("Read and write data on all hotplugged disks!",
+                          logging.info)
+    os_type = params["os_type"]
+    if os_type == "linux":
+        sub_test_type = params.get("sub_test_type", "dd_test")
+        for image in image_list:
+            params["dd_if"] = "ZERO"
+            params["dd_of"] = image
+            utils_test.run_virt_sub_test(test, params, env, sub_test_type)
+
+            params["dd_if"] = image
+            params["dd_of"] = "NULL"
+            utils_test.run_virt_sub_test(test, params, env, sub_test_type)
+    elif os_type == "windows":
+        disk_index = params.objects("disk_index")
+        disk_letter = params.objects("disk_letter")
+        for num in range(image_num):
+            utils_misc.format_windows_disk(session, disk_index[num],
+                                           mountpoint=disk_letter[num])
+            iozone_cmd = params.get("iozone_cmd") % disk_letter[num]
+            iozone_cmd = utils_misc.set_winutils_letter(session, iozone_cmd)
+            status, output = session.cmd_status_output(iozone_cmd,
+                                                       timeout=cmd_timeout)
+            if status:
+                test.fail("Check block device '%s' failed! Output: %s"
+                          % (disk_letter[num], output))
```

Comment:
It is better to notice which block disk failed rather than disk letter.

Reply:
Modified, Thanks!